src/System.ObjectModel/tests/ObservableCollection/ObservableCollection_MethodsTest.cs

```diff
@@ -0,0 +1,843 @@
+                Assert.Equal(anArray.Length, collection.Count);
+            }
+        }
+
+        /// <summary>
+        /// Tests that the appropriate collectionchanged event is fired when
+        /// an item is replaced in the collection.
+        /// </summary>
+        [Fact]
+        public static void ReplaceItemTest()
+        {
+            string[] anArray = new string[] { "one", "two", "three", "four" };
+            ObservableCollection<string> collection = new ObservableCollection<string>((IEnumerable<string>)anArray);
+            CollectionAndPropertyChangedTester helper = new CollectionAndPropertyChangedTester();
+            helper.ReplaceItemTest(collection, 1, "seven");
+            helper.ReplaceItemTest(collection, 3, "zero");
+        }
+
+        /// <summary>
+        /// Tests that contains returns true when the item is in the collection
+        /// and false otherwise.
+        /// </summary>
+        [Fact]
+        public static void ContainsTest()
+        {
+            string[] anArray = new string[] { "one", "two", "three", "four" };
+            ObservableCollection<string> collection = new ObservableCollection<string>((IEnumerable<string>)anArray);
+            string collectionString = "";
+
+            foreach (var item in collection)
+                collectionString += item + ", ";
+
+            for (int i = 0; i < collection.Count; ++i)
+                Assert.True(collection.Contains(anArray[i]), "ObservableCollection did not contain the item: " + anArray[i] + " Collection: " + collectionString);
+
+            string g = "six";
+            Assert.False(collection.Contains(g), "Collection contained an item that should not have been there. guid: " + g + " Collection: " + collectionString);
+            Assert.False(collection.Contains(null), "Collection should not have contained null. Collection: " + collectionString);
+        }
+
+        /// <summary>
+        /// Tests that the index of an item can be retrieved when the item is
+        /// in the collection and -1 otherwise.
+        /// </summary>
+        [Fact]
+        public static void IndexOfTest()
+        {
+            string[] anArray = new string[] { "one", "two", "three", "four" };
+            ObservableCollection<string> collection = new ObservableCollection<string>((IEnumerable<string>)anArray);
+
+            for (int i = 0; i < anArray.Length; ++i)
+                Assert.Equal(i, collection.IndexOf(anArray[i]));
+
+            Assert.Equal(-1, collection.IndexOf("seven"));
+            Assert.Equal(-1, collection.IndexOf(null));
+
+            // testing that the first occurance is the index returned.
+            ObservableCollection<int> intCol = new ObservableCollection<int>();
+            for (int i = 0; i < 4; ++i)
+                intCol.Add(i % 2);
+
+            string collectionString = "";
+            foreach (var item in intCol)
+                collectionString += item + ", ";
```

Comment:
I cleaned up this instance and I will do a pass through the rest of the tests to make sure they are all cleaned up.